src/core/core.scale.js

```diff
@@ -537,6 +541,39 @@ module.exports = Element.extend({
 		return rawValue;
 	},
 
+	_parseValue: function(raw) {
+		var value = +this.getRightValue(raw);
+		var start, end, min, max;
+
+		if (helpers.isArray(value)) {
+			start = value[0];
+			end = value[1];
+			min = Math.min(start, end);
+			max = Math.max(start, end);
+		} else {
+			start = undefined;
+			end = value;
+			min = value;
+			max = value;
+		}
+
+		return {
+			min: min,
+			max: max,
+			start: start,
+			end: end
```

Comment:
I would consider adding a `size` here and using it instead of `value.start === undefined ? value.end : value.max >= 0 && value.min >= 0 ? value.max - value.min : value.min - value.max` in couple of places

Reply:
its possible, but in fact it used only once. And i think because this logic is mainly to bars maybe it should be placed in bar.controller rather then in global scale file

Reply:
I like keeping this simpler and not including `size` actually

Reply:
Its probably usable in too few places to add it here, so lets forget that.